modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ZonePartitionReplicaImpl.java

```diff
@@ -35,17 +35,28 @@ public class ZonePartitionReplicaImpl implements Replica {
 
     private final ReplicaListener listener;
 
+    TopologyAwareRaftGroupService raftClient;
+
+    /**
+     * Constructor.
+     *
+     * @param replicaGrpId  Replication group id.
+     * @param listener Listener for the replica.
+     * @param raftClient Raft client.
+     */
     public ZonePartitionReplicaImpl(
             ReplicationGroupId replicaGrpId,
-            ReplicaListener listener
+            ReplicaListener listener,
+            TopologyAwareRaftGroupService raftClient
     )  {
         this.replicaGrpId = replicaGrpId;
         this.listener = listener;
+        this.raftClient = raftClient;
```

Comment:
as far as I can see, we have todo in `org.apache.ignite.internal.replicator.Replica#raftClient`, so it will be changed once this todo will be resolved

Reply:
The getter will be removed, yes, but raft-client itself should be in listener in my mind. Now we have both in `ReplicaImpl` and `PartitionReplicaListener`, but if we take a look on it's usage, then:
- in replica there only one run on the client with the only `PrimaryReplicaChangeCommand` and `transferLeadership` call, both in `processLeaseGrantedMessage`.
- all logic of raft-client usage are in `ReplicaListener`, mainly `invoke` use it (btw it handles `PrimaryReplicaChangeCommand` too).

Therefore there should be (for me in the future) a duty to remove raft-client from replica in favor of `ReplicaListener`, but not the opposite.

Getter is temporal until the described activity is done, but for now it will be presented.

Reply:
I'll change it after https://issues.apache.org/jira/browse/IGNITE-22459 will be merged, thanks for the comment.